server/mpm/event/event.c

```diff
@@ -1961,13 +2004,21 @@ static void * APR_THREAD_FUNC listener_thread(apr_thread_t * thd, void *dummy)
             num = 0;
         }
 
-        if (listener_may_exit) {
-            close_listeners(&closed);
-            if (terminate_mode == ST_UNGRACEFUL
-                || apr_atomic_read32(&connection_count) == 0)
-                break;
+        if (APLOGtrace7(ap_server_conf)) {
+            now = apr_time_now();
```

Comment:
Is it wise to use `now` here or should we use a new temporary variable here to use it for the `ap_log_error` parameters? Otherwise trace7 logs have side effects.

Reply:
`now` is (re)set to `apr_time_now()` just before it's used (everywhere) in listener loop, `now` before and after the poll() are different anyway. So it's indeed a side affect but just like ap_log_error() itself when the LogLevel is really trace7.
But I agree that the handling of `now` is not very nice in the listener, too many `now = apr_time_now();` (including for the `goto do_maintenance;`, I'll see if I can write this better.